query/influxql/compiler.go

```diff
@@ -34,15 +38,44 @@ func NewCompiler(dbrpMappingSvc platform.DBRPMappingService) *Compiler {
 
 // Compile tranpiles the query into a specification.
 func (c *Compiler) Compile(ctx context.Context) (*flux.Spec, error) {
+	var now time.Time
+	if c.Now != nil {
+		now = *c.Now
+	} else {
+		now = time.Now()
+	}
 	transpiler := NewTranspilerWithConfig(
 		c.dbrpMappingSvc,
 		Config{
 			Cluster:                c.Cluster,
 			DefaultDatabase:        c.DB,
 			DefaultRetentionPolicy: c.RP,
+			Now:                    now,
 		},
 	)
-	return transpiler.Transpile(ctx, c.Query)
+	astPkg, err := transpiler.Transpile(ctx, c.Query)
+	if err != nil {
+		return nil, err
+	}
+
+	semPkg, err := semantic.New(astPkg)
+	if err != nil {
+		return nil, err
+	}
+
+	itrp := interpreter.NewInterpreter()
+	universe := flux.Prelude()
+
+	sideEffects, err := itrp.Eval(semPkg, universe, flux.StdLib())
+	if err != nil {
+		return nil, err
+	}
+
+	spec, err := flux.ToSpec(sideEffects, now)
+	if err != nil {
+		return nil, err
+	}
+	return spec, nil
```

Comment:
I'm not really sure. I was under the impression the compiler needed to produce a spec and how it gets there is irrelevant. @wolffcm is adding code to the flux project proper that adds a compiler implementation that takes a spec so I'll be able to reuse that code instead of copying/pasting here. I think he merged it so I'll ask him to do a follow up to get it here too instead of my quick copy/paste job.